flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/ExecutionFailureHandler.java

```diff
@@ -51,22 +57,35 @@ public class ExecutionFailureHandler {
     /** Number of all restarts happened since this job is submitted. */
     private long numberOfRestarts;
 
+    private final Context taskFailureCtx;
+    private final Context globalFailureCtx;
```

Comment:
IIUC, we we creating two contexts to host different `FailureType`?
This does not seems to be very flexible. Later we need to add one more context for `TASK_MANAGER` type, and maybe even more if more types are introduced.

Also, like @dmvk mentioned in https://github.com/apache/flink/pull/22506#discussion_r1195408929. We may need to provide more information about the error to the enrichers, e.g. related task. Such information can be decided only after an error occurs. So maybe a more flexible way is to create a new context for each failure, i.e. create from `failedExecution ` in `handleFailure ()`.

It's also fine for me to do it in a follow-up task.

Reply:
Agreed, lets use FLINK-32114 to track this -- linked this comment there too!